mariadb: fix invalid PLUGIN_AUTH_PAM value

maxwindiff opened the pull request:
###### Motivation for this change

Fix CMake build error on Mac:

```
CMake Error at cmake/plugin.cmake:108 (MESSAGE):
  Invalid value for PLUGIN_AUTH_PAM
Call Stack (most recent call first):
  plugin/auth_pam/CMakeLists.txt:41 (MYSQL_ADD_PLUGIN)
```

Based on https://mariadb.com/kb/en/specifying-which-plugins-to-build/, valid values of the flag are `YES`,`NO`,`STATIC`,`DYNAMIC`,`AUTO`. I believe the `OFF` is supposed to be `NO`.

###### Things done

I tried building the package locally using `nix-env -i mariadb.server -f $PWD`. It did allow the build process to get pass the configuration phase, but unfortunately I couldn't complete the build as it ran into a compilation error in RocksDB, which seem to be an M1 Mac problem. That said, I think the `PLUGIN_AUTH_PAM` flag still need be changed.

```[ 35%] Building CXX object storage/rocksdb/CMakeFiles/rocksdblib.dir/rocksdb/util/murmurhash.cc.o
/tmp/nix-build-mariadb-server-10.6.3.drv-1/mariadb-10.6.3/storage/rocksdb/rocksdb/util/crc32c.cc:500:18: error: use of undeclared identifier 'isSSE42'
  has_fast_crc = isSSE42();
                 ^
/tmp/nix-build-mariadb-server-10.6.3.drv-1/mariadb-10.6.3/storage/rocksdb/rocksdb/util/crc32c.cc:1230:7: error: use of undeclared identifier 'isSSE42'
  if (isSSE42()) {
      ^
/tmp/nix-build-mariadb-server-10.6.3.drv-1/mariadb-10.6.3/storage/rocksdb/rocksdb/util/crc32c.cc:1231:9: error: use of undeclared identifier 'isPCLMULQDQ'
    if (isPCLMULQDQ()) {
```

maxwindiff committed:
mariadb: fix invalid PLUGIN_AUTH_PAM value

AlexChalk commented:
Is there anything I can do to help get this merged?

aanderse commented:
There are more `OFF` values. Should we fix them too?

AlexChalk commented:
@maxwindiff `nix-shell -p rocksdb` works ok for me on M1, I'll try and figure out how to QA this locally and see if it builds now, but let me know if you get there first :).

maxwindiff commented:
I've brought the PR up to date and made som small changes. Seems to be working:

```
 21:19:34 nixpkgs (fix-mariadb-auth-pam-flag) % nix shell mariadb.server -f $PWD 
Restored session: Thu Feb  3 21:19:34 PST 2022
 21:19:37 nixpkgs (fix-mariadb-auth-pam-flag) % mariadbd --help
mariadbd  Ver 10.6.5-MariaDB for osx10.17 on arm64 (MariaDB Server)
Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Starts the MariaDB database server.

Usage: mariadbd [OPTIONS]

For more help options (several pages), use mysqld --verbose --help.
```

@aanderse - I think the other `OFF`s are valid. The `-DPLUGIN_*` variables seem to follow a different convention.

flmuel commented:
please merge :) maybe you can give me a hint how can I fix this locally? i am using shell.nix configs most of the time

aanderse commented:
Merging because this makes sense.

maxwindiff commented:
Thanks @aanderse, rebased and squashed.

Built again and seems to work (and now on 10.6.7):

```
 19:10:54 nixpkgs (fix-mariadb-auth-pam-flag) % nix shell mariadb.server -f $PWD 
 19:16:38 nixpkgs (fix-mariadb-auth-pam-flag) % mariadbd --help
mariadbd  Ver 10.6.7-MariaDB for osx10.17 on arm64 (MariaDB Server)
Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Starts the MariaDB database server.

Usage: mariadbd [OPTIONS]

For more help options (several pages), use mysqld --verbose --help.
```
